### ISSUE TITLE
RPC calls no longer fail immediately after the server is stopped

### ISSUE THREAD

**Issue by an-p** [grpc/grpc-node#2570]
### Problem description
With grpc-js@1.8.21 and before, RPC calls fail immediately with `14 UNAVAILABLE` after the connection to the server is lost, which seems to be in line with [gRFC A62](https://github.com/grpc/proposal/blob/master/A62-pick-first.md):
> During the backoff period, wait_for_ready RPCs are queued while other RPCs fail.

However, since grpc-js@1.9.0, which implements gRFC A62, RPC calls only fail after exceeding the deadline, despite `waitForReady` being set to false (the default). It looks like the subchannel never enters the TRANSIENT_FAILURE state in the first place.

### Reproduction steps
Send RPC calls periodically, then stop the server.

Example repository to compare the behavior of both grpc-js versions: https://github.com/an-p/grpc-js-bug

### Environment
 - OS name, version and architecture: Ubuntu 22.04, Windows 10 22H2 (both amd64)
 - Node version: 20.6.0
 - Node installation method: nvm
 - If applicable, compiler version: n/a
 - Package name and version: grpc-js@1.9.2

### Additional context

<details>
  <summary>Debug logs</summary>

```
D 2023-09-11T14:40:20.317Z | index | Loading @grpc/grpc-js version 1.9.2
D 2023-09-11T14:40:20.364Z | server | (1) Server constructed
D 2023-09-11T14:40:20.366Z | dns_resolver | Resolver constructed for target dns:127.0.0.1:12345
D 2023-09-11T14:40:20.367Z | dns_resolver | Returning IP address for target dns:127.0.0.1:12345
D 2023-09-11T14:40:20.369Z | resolving_load_balancer | dns:127.0.0.1:12345 IDLE -> IDLE
D 2023-09-11T14:40:20.369Z | connectivity_state | (2) dns:127.0.0.1:12345 IDLE -> IDLE
D 2023-09-11T14:40:20.369Z | dns_resolver | Resolver constructed for target dns:127.0.0.1:12345
D 2023-09-11T14:40:20.369Z | channel | (2) dns:127.0.0.1:12345 Channel constructed with options {
  "grpc.enable_retries": 0
}
D 2023-09-11T14:40:20.370Z | channel_stacktrace | (2) Channel constructed 
    at new InternalChannel (/path/to/cwd/node_modules/@grpc/grpc-js/build/src/internal-channel.js:237:23)
    at new ChannelImplementation (/path/to/cwd/node_modules/@grpc/grpc-js/build/src/channel.js:35:32)
    at new Client (/path/to/cwd/node_modules/@grpc/grpc-js/build/src/client.js:65:36)
    at new ServiceClientImpl (/path/to/cwd/node_modules/@grpc/grpc-js/build/src/make-client.js:58:5)
    at Object.<anonymous> (/path/to/cwd/main.js:39:18)
    at Module._compile (node:internal/modules/cjs/loader:1241:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1295:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
D 2023-09-11T14:40:20.371Z | server | (1) Attempting to bind 127.0.0.1:12345
D 2023-09-11T14:40:20.374Z | server | (1) Successfully bound 127.0.0.1:12345
D 2023-09-11T14:40:20.471Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [0] method="/EchoService/Echo", deadline=2023-09-11T14:40:20.970Z
D 2023-09-11T14:40:20.471Z | resolving_call | [0] Created
D 2023-09-11T14:40:20.471Z | resolving_call | [0] Deadline: 2023-09-11T14:40:20.970Z
D 2023-09-11T14:40:20.472Z | resolving_call | [0] Deadline will be reached in 498ms
D 2023-09-11T14:40:20.472Z | resolving_call | [0] start called
D 2023-09-11T14:40:20.472Z | dns_resolver | Returning IP address for target dns:127.0.0.1:12345
D 2023-09-11T14:40:20.472Z | resolving_load_balancer | dns:127.0.0.1:12345 IDLE -> CONNECTING
D 2023-09-11T14:40:20.472Z | connectivity_state | (2) dns:127.0.0.1:12345 IDLE -> CONNECTING
D 2023-09-11T14:40:20.472Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.ref | configSelectionQueue.length=1 pickQueue.length=0
D 2023-09-11T14:40:20.473Z | resolving_call | [0] startRead called
D 2023-09-11T14:40:20.473Z | resolving_call | [0] write() called with message of length 3
D 2023-09-11T14:40:20.473Z | resolving_call | [0] halfClose called
[GRPC 1.9|0.110] RPC call sent
D 2023-09-11T14:40:20.475Z | subchannel | (4) 127.0.0.1:12345 Subchannel constructed with options {
  "grpc.enable_retries": 0
}
D 2023-09-11T14:40:20.475Z | subchannel_refcount | (4) 127.0.0.1:12345 refcount 0 -> 1
D 2023-09-11T14:40:20.475Z | subchannel_refcount | (4) 127.0.0.1:12345 refcount 1 -> 2
D 2023-09-11T14:40:20.476Z | pick_first | Start connecting to subchannel with address 127.0.0.1:12345
D 2023-09-11T14:40:20.476Z | pick_first | IDLE -> CONNECTING
D 2023-09-11T14:40:20.476Z | resolving_load_balancer | dns:127.0.0.1:12345 CONNECTING -> CONNECTING
D 2023-09-11T14:40:20.476Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.unref | configSelectionQueue.length=1 pickQueue.length=0
D 2023-09-11T14:40:20.476Z | connectivity_state | (2) dns:127.0.0.1:12345 CONNECTING -> CONNECTING
D 2023-09-11T14:40:20.477Z | subchannel | (4) 127.0.0.1:12345 IDLE -> CONNECTING
D 2023-09-11T14:40:20.477Z | transport | dns:127.0.0.1:12345 creating HTTP/2 session to 127.0.0.1:12345
D 2023-09-11T14:40:20.478Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [1] method="/EchoService/Echo"
D 2023-09-11T14:40:20.479Z | resolving_call | [0] Created child [1]
D 2023-09-11T14:40:20.479Z | load_balancing_call | [1] start called
D 2023-09-11T14:40:20.491Z | load_balancing_call | [1] Pick called
D 2023-09-11T14:40:20.491Z | load_balancing_call | [1] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:20.491Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.ref | configSelectionQueue.length=0 pickQueue.length=1
D 2023-09-11T14:40:20.491Z | load_balancing_call | [1] startRead called
D 2023-09-11T14:40:20.492Z | load_balancing_call | [1] write() called with message of length 8
D 2023-09-11T14:40:20.492Z | load_balancing_call | [1] halfClose called
D 2023-09-11T14:40:20.494Z | subchannel | (4) 127.0.0.1:12345 CONNECTING -> READY
D 2023-09-11T14:40:20.494Z | pick_first | Pick subchannel with address 127.0.0.1:12345
D 2023-09-11T14:40:20.494Z | subchannel_refcount | (4) 127.0.0.1:12345 refcount 2 -> 3
D 2023-09-11T14:40:20.494Z | subchannel_refcount | (4) 127.0.0.1:12345 refcount 3 -> 2
D 2023-09-11T14:40:20.495Z | pick_first | CONNECTING -> READY
D 2023-09-11T14:40:20.495Z | resolving_load_balancer | dns:127.0.0.1:12345 CONNECTING -> READY
D 2023-09-11T14:40:20.495Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.unref | configSelectionQueue.length=0 pickQueue.length=0
D 2023-09-11T14:40:20.495Z | load_balancing_call | [1] Pick called
D 2023-09-11T14:40:20.495Z | load_balancing_call | [1] Pick result: COMPLETE subchannel: (4) 127.0.0.1:12345 status: undefined undefined
D 2023-09-11T14:40:20.495Z | connectivity_state | (2) dns:127.0.0.1:12345 CONNECTING -> READY
D 2023-09-11T14:40:20.497Z | transport_flowctrl | (6) 127.0.0.1:12345 local window size: 65535 remote window size: 65535
D 2023-09-11T14:40:20.497Z | transport_internals | (6) 127.0.0.1:12345 session.closed=false session.destroyed=false session.socket.destroyed=false
D 2023-09-11T14:40:20.497Z | load_balancing_call | [1] Created child call [2]
D 2023-09-11T14:40:20.497Z | subchannel_call | [2] write() called with message of length 8
D 2023-09-11T14:40:20.498Z | subchannel_call | [2] sending data chunk of length 8
D 2023-09-11T14:40:20.498Z | subchannel_call | [2] end() called
D 2023-09-11T14:40:20.498Z | subchannel_call | [2] calling end() on HTTP/2 stream
D 2023-09-11T14:40:20.499Z | server | (1) Received call to method /EchoService/Echo at address null
D 2023-09-11T14:40:20.500Z | server_call | Request to /EchoService/Echo received headers {"grpc-accept-encoding":["identity,deflate,gzip"],"accept-encoding":["identity"],"grpc-timeout":["475m"],"user-agent":["grpc-node-js/1.9.2"],"content-type":["application/grpc"],"te":["trailers"]}
D 2023-09-11T14:40:20.502Z | server_call | Request to method /EchoService/Echo ended with status code: OK details: OK
D 2023-09-11T14:40:20.502Z | transport | (6) 127.0.0.1:12345 local settings acknowledged by remote: {"headerTableSize":4096,"enablePush":true,"initialWindowSize":65535,"maxFrameSize":16384,"maxConcurrentStreams":4294967295,"maxHeaderListSize":4294967295,"maxHeaderSize":4294967295,"enableConnectProtocol":false}
D 2023-09-11T14:40:20.502Z | subchannel_call | [2] Received server headers:
		:status: 200
		content-type: application/grpc+proto
		grpc-accept-encoding: identity,deflate,gzip
		grpc-encoding: identity
		date: Mon, 11 Sep 2023 14:40:20 GMT

D 2023-09-11T14:40:20.503Z | load_balancing_call | [1] Received metadata
D 2023-09-11T14:40:20.503Z | resolving_call | [0] Received metadata
D 2023-09-11T14:40:20.503Z | subchannel_call | [2] receive HTTP/2 data frame of length 8
D 2023-09-11T14:40:20.503Z | subchannel_call | [2] parsed message of length 8
D 2023-09-11T14:40:20.503Z | subchannel_call | [2] pushing to reader message of length 8
D 2023-09-11T14:40:20.503Z | load_balancing_call | [1] Received message
D 2023-09-11T14:40:20.504Z | resolving_call | [0] Received message
D 2023-09-11T14:40:20.504Z | resolving_call | [0] Finished filtering received message
D 2023-09-11T14:40:20.504Z | server_call | Request to method /EchoService/Echo stream closed with rstCode 0
D 2023-09-11T14:40:20.505Z | subchannel_call | [2] Received server trailers:
		grpc-status: 0
		grpc-message: OK

D 2023-09-11T14:40:20.505Z | subchannel_call | [2] received status code 0 from server
D 2023-09-11T14:40:20.505Z | subchannel_call | [2] received status details string "OK" from server
D 2023-09-11T14:40:20.505Z | subchannel_call | [2] close http2 stream with code 0
D 2023-09-11T14:40:20.505Z | subchannel_call | [2] ended with status: code=0 details="OK"
D 2023-09-11T14:40:20.506Z | load_balancing_call | [1] Received status
D 2023-09-11T14:40:20.506Z | load_balancing_call | [1] ended with status: code=0 details="OK"
D 2023-09-11T14:40:20.506Z | resolving_call | [0] Received status
D 2023-09-11T14:40:20.506Z | resolving_call | [0] ended with status: code=0 details="OK"
[GRPC 1.9|0.142] RPC call returned: 0
D 2023-09-11T14:40:20.506Z | subchannel_call | [2] HTTP/2 stream closed with code 0
D 2023-09-11T14:40:20.571Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [3] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.071Z
D 2023-09-11T14:40:20.571Z | resolving_call | [3] Created
D 2023-09-11T14:40:20.571Z | resolving_call | [3] Deadline: 2023-09-11T14:40:21.071Z
D 2023-09-11T14:40:20.571Z | resolving_call | [3] Deadline will be reached in 500ms
D 2023-09-11T14:40:20.571Z | resolving_call | [3] start called
D 2023-09-11T14:40:20.571Z | resolving_call | [3] startRead called
D 2023-09-11T14:40:20.571Z | resolving_call | [3] write() called with message of length 3
D 2023-09-11T14:40:20.572Z | resolving_call | [3] halfClose called
[GRPC 1.9|0.208] RPC call sent
D 2023-09-11T14:40:20.572Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [4] method="/EchoService/Echo"
D 2023-09-11T14:40:20.572Z | resolving_call | [3] Created child [4]
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] start called
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] Pick called
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] Pick result: COMPLETE subchannel: (4) 127.0.0.1:12345 status: undefined undefined
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] startRead called
D 2023-09-11T14:40:20.572Z | transport_flowctrl | (6) 127.0.0.1:12345 local window size: 65527 remote window size: 65527
D 2023-09-11T14:40:20.572Z | transport_internals | (6) 127.0.0.1:12345 session.closed=false session.destroyed=false session.socket.destroyed=false
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] Created child call [5]
D 2023-09-11T14:40:20.572Z | load_balancing_call | [4] write() called with message of length 8
D 2023-09-11T14:40:20.572Z | subchannel_call | [5] write() called with message of length 8
D 2023-09-11T14:40:20.572Z | subchannel_call | [5] sending data chunk of length 8
D 2023-09-11T14:40:20.573Z | load_balancing_call | [4] halfClose called
D 2023-09-11T14:40:20.573Z | subchannel_call | [5] end() called
D 2023-09-11T14:40:20.573Z | subchannel_call | [5] calling end() on HTTP/2 stream
D 2023-09-11T14:40:20.573Z | server | (1) Received call to method /EchoService/Echo at address null
D 2023-09-11T14:40:20.573Z | server_call | Request to /EchoService/Echo received headers {"grpc-accept-encoding":["identity,deflate,gzip"],"accept-encoding":["identity"],"grpc-timeout":["499m"],"user-agent":["grpc-node-js/1.9.2"],"content-type":["application/grpc"],"te":["trailers"]}
D 2023-09-11T14:40:20.574Z | server_call | Request to method /EchoService/Echo ended with status code: OK details: OK
D 2023-09-11T14:40:20.574Z | subchannel_call | [5] Received server headers:
		:status: 200
		content-type: application/grpc+proto
		grpc-accept-encoding: identity,deflate,gzip
		grpc-encoding: identity
		date: Mon, 11 Sep 2023 14:40:20 GMT

D 2023-09-11T14:40:20.574Z | load_balancing_call | [4] Received metadata
D 2023-09-11T14:40:20.574Z | resolving_call | [3] Received metadata
D 2023-09-11T14:40:20.574Z | subchannel_call | [5] receive HTTP/2 data frame of length 8
D 2023-09-11T14:40:20.574Z | subchannel_call | [5] parsed message of length 8
D 2023-09-11T14:40:20.574Z | subchannel_call | [5] pushing to reader message of length 8
D 2023-09-11T14:40:20.574Z | load_balancing_call | [4] Received message
D 2023-09-11T14:40:20.574Z | resolving_call | [3] Received message
D 2023-09-11T14:40:20.574Z | resolving_call | [3] Finished filtering received message
D 2023-09-11T14:40:20.575Z | server_call | Request to method /EchoService/Echo stream closed with rstCode 0
D 2023-09-11T14:40:20.575Z | subchannel_call | [5] Received server trailers:
		grpc-status: 0
		grpc-message: OK

D 2023-09-11T14:40:20.575Z | subchannel_call | [5] received status code 0 from server
D 2023-09-11T14:40:20.575Z | subchannel_call | [5] received status details string "OK" from server
D 2023-09-11T14:40:20.575Z | subchannel_call | [5] close http2 stream with code 0
D 2023-09-11T14:40:20.575Z | subchannel_call | [5] ended with status: code=0 details="OK"
D 2023-09-11T14:40:20.575Z | load_balancing_call | [4] Received status
D 2023-09-11T14:40:20.575Z | load_balancing_call | [4] ended with status: code=0 details="OK"
D 2023-09-11T14:40:20.575Z | resolving_call | [3] Received status
D 2023-09-11T14:40:20.575Z | resolving_call | [3] ended with status: code=0 details="OK"
[GRPC 1.9|0.211] RPC call returned: 1
D 2023-09-11T14:40:20.575Z | subchannel_call | [5] HTTP/2 stream closed with code 0
D 2023-09-11T14:40:20.622Z | transport | (6) 127.0.0.1:12345 connection closed by GOAWAY with code 8
D 2023-09-11T14:40:20.622Z | subchannel | (4) 127.0.0.1:12345 READY -> IDLE
D 2023-09-11T14:40:20.623Z | subchannel_refcount | (4) 127.0.0.1:12345 refcount 2 -> 1
D 2023-09-11T14:40:20.623Z | pick_first | READY -> IDLE
D 2023-09-11T14:40:20.623Z | resolving_load_balancer | dns:127.0.0.1:12345 READY -> IDLE
D 2023-09-11T14:40:20.623Z | connectivity_state | (2) dns:127.0.0.1:12345 READY -> IDLE
D 2023-09-11T14:40:20.625Z | transport | (6) 127.0.0.1:12345 session closed
D 2023-09-11T14:40:20.671Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [6] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.171Z
D 2023-09-11T14:40:20.671Z | resolving_call | [6] Created
D 2023-09-11T14:40:20.671Z | resolving_call | [6] Deadline: 2023-09-11T14:40:21.171Z
D 2023-09-11T14:40:20.672Z | resolving_call | [6] Deadline will be reached in 499ms
D 2023-09-11T14:40:20.672Z | resolving_call | [6] start called
D 2023-09-11T14:40:20.672Z | resolving_call | [6] startRead called
D 2023-09-11T14:40:20.672Z | resolving_call | [6] write() called with message of length 3
D 2023-09-11T14:40:20.672Z | resolving_call | [6] halfClose called
[GRPC 1.9|0.308] RPC call sent
D 2023-09-11T14:40:20.672Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [7] method="/EchoService/Echo"
D 2023-09-11T14:40:20.672Z | resolving_call | [6] Created child [7]
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] start called
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] Pick called
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:20.672Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.ref | configSelectionQueue.length=0 pickQueue.length=1
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] startRead called
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] write() called with message of length 8
D 2023-09-11T14:40:20.672Z | load_balancing_call | [7] halfClose called
D 2023-09-11T14:40:20.772Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [8] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.272Z
D 2023-09-11T14:40:20.772Z | resolving_call | [8] Created
D 2023-09-11T14:40:20.772Z | resolving_call | [8] Deadline: 2023-09-11T14:40:21.272Z
D 2023-09-11T14:40:20.772Z | resolving_call | [8] Deadline will be reached in 500ms
D 2023-09-11T14:40:20.773Z | resolving_call | [8] start called
D 2023-09-11T14:40:20.773Z | resolving_call | [8] startRead called
D 2023-09-11T14:40:20.773Z | resolving_call | [8] write() called with message of length 3
D 2023-09-11T14:40:20.773Z | resolving_call | [8] halfClose called
[GRPC 1.9|0.409] RPC call sent
D 2023-09-11T14:40:20.773Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [9] method="/EchoService/Echo"
D 2023-09-11T14:40:20.773Z | resolving_call | [8] Created child [9]
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] start called
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] Pick called
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] startRead called
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] write() called with message of length 8
D 2023-09-11T14:40:20.773Z | load_balancing_call | [9] halfClose called
D 2023-09-11T14:40:20.873Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [10] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.373Z
D 2023-09-11T14:40:20.873Z | resolving_call | [10] Created
D 2023-09-11T14:40:20.873Z | resolving_call | [10] Deadline: 2023-09-11T14:40:21.373Z
D 2023-09-11T14:40:20.873Z | resolving_call | [10] Deadline will be reached in 500ms
D 2023-09-11T14:40:20.873Z | resolving_call | [10] start called
D 2023-09-11T14:40:20.873Z | resolving_call | [10] startRead called
D 2023-09-11T14:40:20.874Z | resolving_call | [10] write() called with message of length 3
D 2023-09-11T14:40:20.874Z | resolving_call | [10] halfClose called
[GRPC 1.9|0.510] RPC call sent
D 2023-09-11T14:40:20.874Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [11] method="/EchoService/Echo"
D 2023-09-11T14:40:20.874Z | resolving_call | [10] Created child [11]
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] start called
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] Pick called
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] startRead called
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] write() called with message of length 8
D 2023-09-11T14:40:20.874Z | load_balancing_call | [11] halfClose called
D 2023-09-11T14:40:20.974Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [12] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.474Z
D 2023-09-11T14:40:20.974Z | resolving_call | [12] Created
D 2023-09-11T14:40:20.974Z | resolving_call | [12] Deadline: 2023-09-11T14:40:21.474Z
D 2023-09-11T14:40:20.974Z | resolving_call | [12] Deadline will be reached in 500ms
D 2023-09-11T14:40:20.974Z | resolving_call | [12] start called
D 2023-09-11T14:40:20.974Z | resolving_call | [12] startRead called
D 2023-09-11T14:40:20.974Z | resolving_call | [12] write() called with message of length 3
D 2023-09-11T14:40:20.974Z | resolving_call | [12] halfClose called
[GRPC 1.9|0.611] RPC call sent
D 2023-09-11T14:40:20.975Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [13] method="/EchoService/Echo"
D 2023-09-11T14:40:20.975Z | resolving_call | [12] Created child [13]
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] start called
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] Pick called
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] startRead called
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] write() called with message of length 8
D 2023-09-11T14:40:20.975Z | load_balancing_call | [13] halfClose called
D 2023-09-11T14:40:21.074Z | channel | (2) dns:127.0.0.1:12345 createResolvingCall [14] method="/EchoService/Echo", deadline=2023-09-11T14:40:21.574Z
D 2023-09-11T14:40:21.074Z | resolving_call | [14] Created
D 2023-09-11T14:40:21.074Z | resolving_call | [14] Deadline: 2023-09-11T14:40:21.574Z
D 2023-09-11T14:40:21.074Z | resolving_call | [14] Deadline will be reached in 500ms
D 2023-09-11T14:40:21.074Z | resolving_call | [14] start called
D 2023-09-11T14:40:21.075Z | resolving_call | [14] startRead called
D 2023-09-11T14:40:21.075Z | resolving_call | [14] write() called with message of length 3
D 2023-09-11T14:40:21.075Z | resolving_call | [14] halfClose called
[GRPC 1.9|0.711] RPC call sent
D 2023-09-11T14:40:21.075Z | channel | (2) dns:127.0.0.1:12345 createLoadBalancingCall [15] method="/EchoService/Echo"
D 2023-09-11T14:40:21.075Z | resolving_call | [14] Created child [15]
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] start called
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] Pick called
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] startRead called
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] write() called with message of length 8
D 2023-09-11T14:40:21.075Z | load_balancing_call | [15] halfClose called
D 2023-09-11T14:40:21.171Z | resolving_call | [6] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.171Z | load_balancing_call | [7] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.171Z | load_balancing_call | [7] ended with status: code=4 details="Deadline exceeded"
D 2023-09-11T14:40:21.171Z | resolving_call | [6] Received status
D 2023-09-11T14:40:21.171Z | resolving_call | [6] ended with status: code=4 details="Deadline exceeded"
[GRPC 1.9|0.808] RPC call failed: 4 DEADLINE_EXCEEDED: Deadline exceeded
D 2023-09-11T14:40:21.273Z | resolving_call | [8] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.273Z | load_balancing_call | [9] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.273Z | load_balancing_call | [9] ended with status: code=4 details="Deadline exceeded"
D 2023-09-11T14:40:21.273Z | resolving_call | [8] Received status
D 2023-09-11T14:40:21.273Z | resolving_call | [8] ended with status: code=4 details="Deadline exceeded"
[GRPC 1.9|0.909] RPC call failed: 4 DEADLINE_EXCEEDED: Deadline exceeded
D 2023-09-11T14:40:21.374Z | resolving_call | [10] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.374Z | load_balancing_call | [11] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.374Z | load_balancing_call | [11] ended with status: code=4 details="Deadline exceeded"
D 2023-09-11T14:40:21.374Z | resolving_call | [10] Received status
D 2023-09-11T14:40:21.374Z | resolving_call | [10] ended with status: code=4 details="Deadline exceeded"
[GRPC 1.9|1.010] RPC call failed: 4 DEADLINE_EXCEEDED: Deadline exceeded
D 2023-09-11T14:40:21.473Z | resolving_load_balancer | dns:127.0.0.1:12345 IDLE -> CONNECTING
D 2023-09-11T14:40:21.473Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.unref | configSelectionQueue.length=0 pickQueue.length=0
D 2023-09-11T14:40:21.473Z | load_balancing_call | [13] Pick called
D 2023-09-11T14:40:21.473Z | load_balancing_call | [13] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:21.473Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.ref | configSelectionQueue.length=0 pickQueue.length=1
D 2023-09-11T14:40:21.473Z | load_balancing_call | [15] Pick called
D 2023-09-11T14:40:21.473Z | load_balancing_call | [15] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-09-11T14:40:21.473Z | connectivity_state | (2) dns:127.0.0.1:12345 IDLE -> CONNECTING
D 2023-09-11T14:40:21.475Z | resolving_call | [12] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.475Z | load_balancing_call | [13] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.475Z | load_balancing_call | [13] ended with status: code=4 details="Deadline exceeded"
D 2023-09-11T14:40:21.475Z | resolving_call | [12] Received status
D 2023-09-11T14:40:21.475Z | resolving_call | [12] ended with status: code=4 details="Deadline exceeded"
[GRPC 1.9|1.112] RPC call failed: 4 DEADLINE_EXCEEDED: Deadline exceeded
D 2023-09-11T14:40:21.575Z | resolving_call | [14] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.575Z | load_balancing_call | [15] cancelWithStatus code: 4 details: "Deadline exceeded"
D 2023-09-11T14:40:21.575Z | load_balancing_call | [15] ended with status: code=4 details="Deadline exceeded"
D 2023-09-11T14:40:21.575Z | resolving_call | [14] Received status
D 2023-09-11T14:40:21.575Z | resolving_call | [14] ended with status: code=4 details="Deadline exceeded"
[GRPC 1.9|1.211] RPC call failed: 4 DEADLINE_EXCEEDED: Deadline exceeded
D 2023-09-11T14:40:22.823Z | resolving_load_balancer | dns:127.0.0.1:12345 CONNECTING -> IDLE
D 2023-09-11T14:40:22.823Z | channel | (2) dns:127.0.0.1:12345 callRefTimer.unref | configSelectionQueue.length=0 pickQueue.length=0
D 2023-09-11T14:40:22.823Z | connectivity_state | (2) dns:127.0.0.1:12345 CONNECTING -> IDLE
``` 
</details>

**Comment by murgatroid99** [comment]
Here's what I think is happening: when the connection drops, pick_first goes into IDLE and requests DNS reresolution, which would in turn trigger a new connection attempt. However, it runs into the DNS resolver's rate limit, so it takes some time to start connecting again. This is a consequence of the short timespan the test is operating on: in practical usage, it is unusual for connections to drop that quickly after they are established. The DNS resolver's rate limit defaults to 30 seconds, and is controlled by the `grpc.dns_min_time_between_resolutions_ms` option. Changing that option to a much smaller value does produce different behavior, but it looks like there's a different bug that makes it still take half a second or so to reach the expected state.

I also want to note that you have misunderstood gRFC A62. We have different handling for failing to establish a connection, and for an already established connection getting dropped. The gRFC deals with the first case, and your test deals with the second, so the sentence you quoted doesn't apply to this situation.

**Comment by an-p** [comment]
Thank you for the clarification!

We're only using IP addresses, so I did not expect DNS resolution to cause a significant delay.
I can confirm that the issue is fixed by the changes in #2571 (or by reducing `grpc.dns_min_time_between_resolutions_ms` as a workaround).

**Comment by murgatroid99** [comment]
If you are only using IP addresses in production, you can also use the `ipv4` or `ipv6` scheme as documented [here](https://github.com/grpc/grpc/blob/master/doc/naming.md). Those use a different code path that doesn't have any rate limit.

**Comment by murgatroid99** [comment]
The fixes in #2571 are now out in version 1.9.3.
